### PR TITLE
[WiP] Added LocalStorage adapter.

### PR DESCRIPTION
**Pull request by n1k0**
This is ugly as hell but works as a proof of concept. Will pursue next week.

### CODE REVIEW

File: src/adapters/localStorage.js [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  root.localStorage = require("localStorage");
+}
+
+export default class LocalStorage extends BaseAdapter {
+  constructor(dbname) {
+    super();
+    this._db = null;
+    // public properties
+    this.dbname = dbname;
+    this._keyStoreName = `${this.dbname}/__keys`;
```
Comment (ametaireau) [inline src/adapters/localStorage.js]: Is this key a public one? (it's prefixed by `_`)

File: src/adapters/localStorage.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+  /**
+   * Retrieve a record by its primary key from the localStorage datastore.
+   *
+   * @param  {String} id The record id.
+   * @return {Promise}
+   */
+  get(id) {
+    return Promise.resolve(
+      JSON.parse(localStorage.getItem(`${this.dbname}/${id}`)) || undefined);
```
Comment (ametaireau) [inline src/adapters/localStorage.js]: Can ids be any type of value? (asking for the formatting, in order to avoid collision of names

**Comment by ametaireau** [comment]
I really like to have different adapters, thanks for doing this.

The code looks good to me. r+ (with comments)

File: src/adapters/localStorage.js (hunk 2 above)
Comment (Natim) [inline src/adapters/localStorage.js]: At the moment ids can only be UUID4.

**Pull request merged**